zio-config/src/main/scala/zio/config/PropertyType.scala

```diff
@@ -7,73 +7,73 @@ import zio.config.ReadError.ParseError
 import scala.util.{ Failure, Success, Try }
 
 trait PropertyType[A] {
-  def read(path: String, propertyValue: String): Either[ReadError, A]
+  def read(path: String, propertyValue: String): Either[ReadError[String, String], A]
```

Comment:
Yes, these must all be specialized to string here.

Of course, we can push `K` and `V` into `PropertyType`, and we probably should...